src/ray/core_worker/experimental_mutable_object_manager.cc

```diff
@@ -315,13 +320,20 @@ Status MutableObjectManager::ReadAcquire(const ObjectID &object_id,
   auto timeout_point = ToTimeoutPoint(timeout_ms);
   bool locked = false;
   bool expired = false;
-  do {
-    RAY_RETURN_NOT_OK(object->header->CheckHasError());
-    // The channel is still open. This lock ensures that there is only one reader
-    // at a time. The lock is released in `ReadRelease()`.
-    locked = channel->lock->try_lock();
-    expired = timeout_point && std::chrono::steady_clock::now() >= *timeout_point;
-  } while (!locked && !expired);
+  {
+    std::signal(SIGINT, SignalHandler);
```

Comment:
hmm, this is called every time `ReadAcquire()` is called. I think we should do it only once.

Reply:
Only doing this once would mean placing the signal handler on the process, which is different from normal ray.get, which only checks for signals on the waiting while loops. I'm not totally sure why it does that though?

Reply:
Which loop of normal ray.get were you referring to?

Reply:
https://github.com/ray-project/ray/blob/cdaeaaef672a49568eed515f097cc068958a648b/src/ray/core_worker/store_provider/memory_store/memory_store.cc#L355-L363

from here

Reply:
I think we should try to use the same mechanism `check_signals_`, we can discuss more offline.

Reply:
using check_signals now